cupy/_environment.py

```diff
@@ -369,3 +371,47 @@ def _preload_warning(lib, exc):
             lib=lib, exc_type=type(exc).__name__, exc=str(exc),
             cuda=config['cuda'])
         warnings.warn(msg)
+
+
+def _detect_duplicate_installation():
+    # List of all CuPy packages, including out-dated ones.
+    known = set([
+        'cupy',
+        'cupy-cuda80',
+        'cupy-cuda90',
+        'cupy-cuda91',
+        'cupy-cuda92',
+        'cupy-cuda100',
+        'cupy-cuda101',
+        'cupy-cuda102',
+        'cupy-cuda110',
+        'cupy-cuda111',
+        'cupy-cuda112',
+        'cupy-rocm-4-0',
+    ])
+    installed = set(
+        [p.project_name for p in pkg_resources.working_set])
+    cupy_installed = known & installed
+    if 1 < len(cupy_installed):
+        warnings.warn('''
+--------------------------------------------------------------------------------
+
+  CuPy may not function correctly because multiple CuPy packages are installed
+  in your environment:
+
+    {cupy_installed}
```

Comment:
Could you use an f-string instead of `format`?

Reply:
Ok, I forgot that rule I just added... 😅